code/game/machinery/fabricators/modular_fabricator.dm

```diff
@@ -467,14 +467,28 @@
 		return
 	begin_process()
 
-/obj/machinery/modular_fabricator/proc/make_item(power, var/list/materials_used, var/list/picked_materials, multiplier, coeff, is_stack)
+/obj/machinery/modular_fabricator/proc/make_item(power, var/list/materials_used, var/list/picked_materials, multiplier, coeff, is_stack, requested_design_id, queue_data)
 	if(QDELETED(src))
 		return
 	//Stops the queue
 	if(disabled)
 		operating = FALSE
+		busy = FALSE
+		set_default_sprite()
+		// requeue the item
+		add_to_queue(item_queue, requested_design_id, stored_item_amount + 1, queue_data["build_mat"])
+		stored_item_amount = 0
 		return
 	var/datum/component/material_container/materials = get_material_container()
+	if(!materials.has_materials(materials_used))
+		operating = FALSE
+		wants_operate = TRUE
+		busy = FALSE
+		set_default_sprite()
+		// requeue the item
+		add_to_queue(item_queue, requested_design_id, stored_item_amount + 1, queue_data["build_mat"])
```

Comment:
I would assume this enqueues the item, when you'd need to put it first in queue instead.

Reply:
the code is actually terrible, that's not what it does

i tested it, it works as is, as expected.

this is how the item is queued above